ingestion_tools/scripts/importers/identified_object.py

```diff
@@ -0,0 +1,118 @@
+
+from typing import Any
+
+import pandas as pd
+
+from common.config import DepositionImportConfig
+from common.finders import DefaultImporterFactory
+from common.id_helper import IdentifierHelper
+from common.metadata import IdentifiedObjectMetadata
+from importers.base_importer import BaseFileImporter
+
+
+class IdentifiedObjectIdentifierHelper(IdentifierHelper):
+    @classmethod
+    def _get_container_key(cls, config: DepositionImportConfig, parents: dict[str, Any], *args, **kwargs) -> str:
+        return "-".join(["identified_object", parents["run"].get_output_path()])
+
+    @classmethod
+    def _get_metadata_glob(cls, config, parents, *args, **kwargs) -> str:
+        run = parents["run"]
+        metadata_glob = config.resolve_output_path("identified_object_metadata", run)
+        return metadata_glob
+
+    @classmethod
+    def _generate_hash_key(
+        cls,
+        container_key: str,
+        metadata: dict[str, Any],
+        parents: dict[str, Any],
+        *args,
+        **kwargs,
+    ) -> str:
+        return "-".join(
+            [
+                container_key,
+                str(metadata.get("deposition_id", int(parents["deposition"].name))),
+                str(metadata.get("run_id", parents["run"].name)),
+                str(metadata.get("identified_object_id", "unknown")),
+                str(metadata.get("identified_object_name", "unknown")),
+                str(metadata.get("identified_object_state", "unknown")),
+                str(metadata.get("identified_object_description", "unknown")),
+            ],
+        )
+
+
+class IdentifiedObjectImporter(BaseFileImporter):
+    type_key = "identified_object"
+    plural_key = "identified_objects"
+    finder_factory = DefaultImporterFactory
+    has_metadata = True
+    dir_path = "{dataset_name}/{run_name}/IdentifiedObjects"
+    metadata_path = "{dataset_name}/{run_name}/IdentifiedObjects/identified_objects_metadata.json"
+
+    def __init__(
+        self,
+        config: DepositionImportConfig,
+        metadata: dict[str, Any],
+        name: str,
+        path: str,
+        allow_imports: bool,
+        parents: dict[str, Any],
+    ):
+        super().__init__(
+            config=config,
+            metadata=metadata,
+            name=name,
+            path=path,
+            allow_imports=allow_imports,
+            parents=parents,
+        )
+        self.identifier = IdentifiedObjectIdentifierHelper.get_identifier(
+            config, self.get_base_metadata(), self.parents,
+        )
+
+    def import_item(self) -> None:
+        if not self.is_import_allowed():
+            print(f"Skipping import of {self.name}")
+            return
+
+        dest_path = self.get_output_path()
+        self.config.fs.makedirs(dest_path)
```

Comment:
This is not needed. As it should already be done by: https://github.com/chanzuckerberg/cryoet-data-portal-backend/blob/a193e7c3f586b2fc1da983422538774c0d68a5d9/ingestion_tools/scripts/common/config.py#L237